src/ess/reduce/nexus/workflow.py

```diff
@@ -465,6 +514,36 @@ def assemble_monitor_data(
     return MonitorData[RunType, MonitorType](_add_variances(da))
 
 
+def parse_disk_choppers(
+    choppers: AllNeXusComponents[snx.NXdisk_chopper, RunType],
+) -> Choppers[RunType]:
+    """Convert the NeXus representation of a chopper to ours."""
+    return Choppers[RunType](
+        sc.DataGroup(
+            {
+                name: extract_chopper_from_nexus(
+                    nexus.compute_component_position(chopper)
+                )
+                for name, chopper in choppers.items()
+            }
+        )
```

Comment:
Can this use https://scipp.github.io/generated/classes/scipp.DataGroup.html#scipp.DataGroup.apply? Feel free to ignore this comment unless shorter.

Reply:
Yes. But it only saves 1 line here. It's better for the analysers, though.